Project 4/SPARQL-JV.md

```diff
@@ -0,0 +1,389 @@
+      ?facility rdf:type touronot:Lighthouse ;
+            touronto:located_in touronto:LowerPeninsulaMichigan ;
+            touronto:nearby_location touronto:LakeMichigan .
+}
+```
+
+**Problem 7-5.**
+- The National Park and Recreation Department is concerned about recent adverse changes in natural habitat for the ruby-throated thrush. Find all parks that may be affected by these conservation efforts.
+- Return all parks that are the natural habits of ruby-throated thrush.
+```
+PREFIX ro: <http://purl.obolibrary.org/obo/ro.owl>
+PREFIX touronto: <https://americathebeautiful.com/tours/ontology/>
+
+SELECT ?park
+WHERE {
+      ?park ro:located_in bfo:site ;
+      bfo:site touronto:contains_habitat touronto:naturalHabitat ;
+      touronto:naturalHabitat touronto:habitat_of zoo:rubyThroatedThrush .
+}
+```
+
+**Problem 7-6.**
+- Find all cameras manufacturerd by Fujifilm (ontocam:manufacturer), released in the year 2020, and has a bluetooth connection function.
+- Return camera name (ontocam:productName), model number (ontocam:modelValue), release date (ontocam:releaseDate).
+```
+PREFIX ontocam: <https://aperturescience.org/onto/>
+PREFIX ro: <http://purl.obolibrary.org/obo/ro.owl>
+
+SELECT ?productName ?modelValue ?releaseDate
+WHERE {
+      ?device ontocam:manufactured_by ontocam:Fujifilm ;
+      ?device rdfs:type ontocam:Camera ;
+      ?device ontocam:released_in_year "2020" ;
+      ?device ro:has_function ontocam:Bluetooth .
+}
+```
+
+**Problem 7-7.**
+- A catalog should display all lighting fixture models that have a multi-brightness function and can be connected as an Internet of Things device (IoT).
+- Return the product name, product model, current price, and current inventory that fit this criteria.
+
+```
+PREFIX ro: <http://purl.obolibrary.org/obo/ro.owl>
+PREFIX homedev: <http://homeandgarden.com/ontology/>
+
+SELECT ?productName ?productModelValue ?productPriceUSD ?inventoryValue
+WHERE {
+        ?product ro:instance_of homedev:LightingFixture ;
+        ?product ro:has_function homedev:multiSettingBrightness ;
+        ?product ro:has_function homedev:internetOfThingsConnection .
+}
+```
+
+**Problem 7-8.**
+- A sample ballot for a local county should display all candidates who are running in an upcoming election.
+- Return the candidate name, candidate's political affiliation, and candidate's website address.
+
+```
+borked
+PREFIX ro: <http://purl.obolibrary.org/obo/ro.owl>
+PREFIX homedev: <http://homeandgarden.com/ontology/>
+
+SELECT ?productName ?productModelValue ?productPriceUSD ?inventoryValue
+WHERE {
+        ?product ro:instance_of homedev:LightingFixture ;
+        ?product ro:has_function homedev:multiSettingBrightness ;
+        ?product ro:has_function homedev:internetOfThingsConnection .
+}
+```
+
+**Problem 7-9.**
+- find all persons who have a passport issued, the person has not left the country, and the person is over the age of 25
+- borked: to be filled in
+```
+borked
+```
+
+
+**Problem 6-1.**
```

Comment:
Isn't this a bit easy compared to, say, 6.5?

Reply:
I added an ordering, additional criteria, and other details. I added the namespace schema guidance.